library/timesync_provider.sh

```diff
@@ -0,0 +1,57 @@
+#!/bin/bash
+# WANT_JSON
+
+is_service_enabled() {
+	local name=$1 runlevel prev_runlevel
+
+	read -r prev_runlevel runlevel < <(runlevel)
+
+	systemctl is-enabled $name.service &> /dev/null || \
+		chkconfig --list $name 2>/dev/null | grep -q "$runlevel:on"
+}
+
+is_service_active() {
+	local name=$1
+
+	systemctl is-active $name.service &> /dev/null || \
+		service $name status &>/dev/null
+}
+
+get_current_ntp_providers() {
+	is_service_active chronyd || is_service_enabled chronyd && echo chrony
+	is_service_active ntpd || is_service_enabled ntpd && echo ntp
+}
+
+get_default_ntp_provider() {
```

Comment:
I would omit this function and do the default provider detection in the task file as before, for several reasons.

1. Consistency - there are several other places in linux-system-roles that check `ansible_distribution` or `ansible_distribution_version`, but nowhere else one parses `/etc/os-release`.
2. The module should do one thing and do it well. In this form the module does several thing: 1. checks for the OS default 2. checks for the current situation 3. implements the algorithm of choosing between them. I would prefer the module to perform only the second - checking the current provider - as this seems to be difficult to do without a module (for the record, as you found out, the `service_facts` module is currently insufficient for the task - it does not return enabled/disabled status of services and also is still _preview_ which is not ideal, so one needs to implement a module for the task). The other parts are easy to do in the task file so it would be better to do them in the task file.
3. Separating the OS default detection from the current service detection would enable the user to do
```yaml
ntp_provider: "{{ ntp_provider_default }}"
```
to always set the OS default regardless of the previous state. This is a reasonable thing to want and it is not possible in the current situation.

Reply:
I implemented the suggested changes in #13